### ISSUE TITLE
2.5.8 Target Size (Minimum) - Some updates needed

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#258]
This is to pull out the remaining work from #80.  Issue 80 was to create a draft for the SC which was incorporated into the FPWD of WCAG2ICT which I'd like to close. This issue will focus on the work that remains for this SC, namely:
- [Phil's reminder comment](https://github.com/w3c/wcag2ict/issues/80#issuecomment-1634428098), to be turned into a review-worthy proposal: 
>Notes 1 and 2 need rework.
>Need to say something along the lines of closed products - there is a large variety in display sizes and a lack of pixel density independent measurements, thus a physical measurement may be more useful.
>However, this should be defined in such a way as to still be feasible for devices with very small screens.
- [Mary Jo's comment](https://github.com/w3c/wcag2ict/issues/80#issuecomment-1634543924) that we need a bullet in the SC Problematic for Closed Functionality section. Copying in the thoughts from the Google spreadsheet Closed Functionality analysis of the SC:
>
>Phil: Problematic due to using CSS pixels as previously discussed in reflow. Physical equivalents would greatly help in implementing this type of requirement. Similarly, testing may not be feasible by any other than the content creator
>
>Mary Jo: Agree, that closed functionality would more likely to not have any method of measuring size of targets using software, would likely not have any concept of device-independent pixels or CSS pixels, and often would be tied to a particular display/configuration (so the pixel density wouldn't be changeable). If there was a concept of pixel density-independent measures, recommend using that. Otherwise, a physical size fallback based on the research for fingertip size could be developed by a standards organization.

**Comment by samogami** [comment]
I would recommend that we make the SC not applicable if there is no CSS, device, or density independent pixels.
Context of use, known information becomes increasingly variable and not event known when applying the SC to all non Web software. 
For MS, Android, Apple devices it can be applicable with some more changes but there are many devices that are not in the big three ecosystem.  

Change Note 2 to:
Note 2: Non-web software that either doesn't have the concept of device or density independent pixels or measurements by pixels,  and the viewing angle and viewing distance is known a [reference pixel](https://www.w3.org/TR/css3-values/#reference-pixel) can be calculate for the reference pixel size.
 * Software that runs on specific hardware, like kiosks or office equipment, where the author knows the physical screen size and, potentially, the pixel density.

Add an additional note:
Note 3: Where Non-web software that either doesn't have the concept of device or density independent pixels, measurements by pixels, or physical measurements are unknown Success Criteria 2.5.8 is not applicable.
 * Software, like TV streaming apps, where the author does not know the physical screen size or viewing distance.

**Comment by pday1** [comment]
I've attempted to collate some of the discussion in the comments of previous issue #80, including Mitch's notes, and Sam's comments on those.

I propose we add the following notes to the existing [Applying SC 2.5.8 Target Size (Minimum) to Non-Web Documents and Software:](https://w3c.github.io/wcag2ict/#applying-sc-2-5-8-target-size-minimum-to-non-web-documents-and-software)section, replacing the existing Note 4. This will then mean that Notes 1-3 of the existing text remain, and we add notes 4-6. 


Non-web software

NOTE 4: Non-web software and their accompanying platform software do not use measurements of CSS pixels. Therefore, platform-specific pixel density-independent measurements should be used. Examples include: px in iOS and MacOS, density-independent pixels (DP) for Android, and device-independent pixels for Windows.

NOTE 5: Non-web software that either doesn't have the concept of device or density independent pixels or measurements by pixels, should use the formula contained in the definition of a [reference pixel](https://www.w3.org/TR/css3-values/#reference-pixel) to calculate the arc-length of the viewing angle using the optimal viewing distance from the screen to calculate the reference pixel size.
Examples include:
	• Software that runs on specific hardware, like kiosks or office equipment, where the author knows the physical screen size and, potentially, the pixel density.

NOTE 6: Non-web software that does not have the concept of device or density independent pixels, measurements by pixels, or allow for physical measurements may require a  physical size fallback based on the research for fingertip size to be developed by a standards organization. See also the discussion on Closed Functionality
Examples include 
	• Software, like TV streaming apps, where the author does not know the physical screen size or viewing distance.

**Comment by GreggVan** [comment]
nice work !!

PS  editorial  only -   
Therefore, platform-specific pixel density-independent measurements should be used.   
should be  (with comma and hyphen added) 
Therefore, platform-specific, pixel-density-independent measurements should be used.

**Comment by maryjom** [comment]
[Survey](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-target-size-notes/) opened to review this.

**Comment by mraccess77** [comment]
I don't have permission to answer the survey but I have a comment to share below:

The "in the same non-web document or software"" bit is problematic because it allows the alternative to be anywhere in the software such as in a completely different screen.  I would like to see something like "in the same context" be added to cover this otherwise it would seem to open up a large loop hole.

**Comment by maryjom** [comment]
@mraccess77 @patrickhlauke The survey is not an AG WG survey but a WCAG2ICT TF survey which is why you are unable to answer. In our review of survey results, I always also look at comments on the related issue that introduces the content.

**Comment by pday1** [comment]
After an animated discussion in the [survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-target-size-notes/results) 
and subsequent task force meeting on December 7th further changes were needed (in particular, much content was similar to content already in the definition of CSS pixel, so this should be referenced and consistent language used). 



**Proposed new content**
I propose we add the following note to the existing [Applying SC 2.5.8 Target Size (Minimum) to Non-Web Documents and Software](#applying-sc-2-5-8-target-size-minimum-to-non-web-documents-and-software) section, replacing the existing Note 4. This will then mean that Notes 1-3 of the existing text remain, and we replace note 4. 

Content has also been suggested for the relevant bullet in the success criteria problematic for closed functionality section, tracked in issue #293 

Non-web software

Note 4 
In technologies where CSS is not used, the definition of 'CSS pixel' still applies as described in [Applying “CSS pixel” to Non-Web Documents and Software](#applying-css-pixel-to-non-web-documents-and-software). 

See also the discussion on [Closed Functionality](#closed-functionality).

**Comment by pday1** [comment]
See also the related discussion in https://github.com/w3c/wcag/issues/3598#issuecomment-1896256742

**Comment by pday1** [comment]
SC problematic for closed: additional content has been added. 

See issue https://github.com/w3c/wcag2ict/issues/293

**Comment by maryjom** [comment]
I'm going to close this as a duplicate of #293 where the new content is being developed.
